test/e2e/operator_test.go

```diff
@@ -624,45 +641,56 @@ func TestAWSLoadBalancerControllerWithWAFRegional(t *testing.T) {
 		waitForDeletion(t, kubeClient, echoNs, defaultTimeout)
 	}()
 
-	wafClient := waf.NewFromConfig(cfg)
+	var webACLID string
+	if !isOnROSA() {
+		wafClient := waf.NewFromConfig(cfg)
 
-	token, err := wafClient.GetChangeToken(context.TODO(), &waf.GetChangeTokenInput{})
-	if err != nil {
-		t.Fatalf("failed to get change token for waf regional classic %v", err)
-	}
-	acl, err := wafClient.CreateWebACL(context.TODO(), &waf.CreateWebACLInput{
-		DefaultAction: &waftypes.WafAction{Type: waftypes.WafActionTypeBlock},
-		MetricName:    aws.String("echoserverclassicacl"),
-		Name:          aws.String("echoserverclassicacl"),
-		ChangeToken:   token.ChangeToken,
-	})
-	if err != nil {
-		t.Fatalf("failed to create aws waf regional acl due to %v", err)
-	}
-	defer func() {
 		token, err := wafClient.GetChangeToken(context.TODO(), &waf.GetChangeTokenInput{})
 		if err != nil {
 			t.Fatalf("failed to get change token for waf regional classic %v", err)
 		}
-
-		_, err = wafClient.DeleteWebACL(context.TODO(), &waf.DeleteWebACLInput{
-			ChangeToken: token.ChangeToken,
-			WebACLId:    acl.WebACL.WebACLId,
+		acl, err := wafClient.CreateWebACL(context.TODO(), &waf.CreateWebACLInput{
+			DefaultAction: &waftypes.WafAction{Type: waftypes.WafActionTypeBlock},
+			MetricName:    aws.String("echoserverclassicacl"),
+			Name:          aws.String("echoserverclassicacl"),
+			ChangeToken:   token.ChangeToken,
 		})
 		if err != nil {
-			t.Fatalf("failed to delete aws waf regional acl due to %v", err)
+			t.Fatalf("failed to create aws waf regional acl due to %v", err)
 		}
-	}()
+		webACLID = *acl.WebACL.WebACLId
```

Comment:
Could acl ever be nil?

Reply:
No, [wafregional.CreateWebACL method](https://github.com/aws/aws-sdk-go-v2/blob/service/wafregional/v1.13.6/service/wafregional/api_op_CreateWebACL.go#L53) returns `nil` only if there is an error. We fail the test in case the error is returned.